## BLAST

[BLAST](https://blast.ncbi.nlm.nih.gov/Blast.cgi)

<img src="Images/BLASTP.png" width="700">

Protein BLAST (blastp)
- Why blastp ? Because evolutionary comparisons are usually more accurate with proteins (why?)
another reason???

- Database :nr (Non‑redundant protein sequences)
- Organism : Vertebrata

- Limiting the analysis to Vertebrates (taxid:7742) ensures that the evolutionary comparison is more focused, allowing us to better observe the conservation of the APOE protein across closely related species.

 the evolutionary conservation of the APOE protein
a BLASTp search was performed against the Vertebrata database. 
The human reference sequence (NP_000032.1) was used as the query.

<img src="Images/blast.png" width="700">

In [4]:
import pandas as pd

# create a dictionary of blastp results
blast_data = {
    'Species': ['Human', 'Bornean orangutan', 'Western lowland gorilla', 'Chimpanzee'],
    'Scientific Name' : ['Homo sapiens', 'Pongo pygmaeus', 'Gorilla gorilla gorilla', 'Pan troglodytes'],
    'Accession': ['NP_000032.1', 'Q9GLM7.1', 'Q9GLM8.1', 'NP_001009007.1'],
    'Identity (%)': [100.00, 97.48, 97.48, 97.16],
    'Query Cover (%)': [100, 100, 100, 100],
    'E-value': [0.0, 0.0, 0.0, 0.0]
}

# Turn the disctionary into the dataframe
df_blast = pd.DataFrame(blast_data)
df_blast.style.set_caption("Description table of BLASTp Results for APOE Protein")


,Species,Scientific Name,Accession,Identity (%),Query Cover (%),E-value
0,Human,Homo sapiens,NP_000032.1,100.000000,100,0.000000
1,Bornean orangutan,Pongo pygmaeus,Q9GLM7.1,97.480000,100,0.000000
2,Western lowland gorilla,Gorilla gorilla gorilla,Q9GLM8.1,97.480000,100,0.000000
3,Chimpanzee,Pan troglodytes,NP_001009007.1,97.160000,100,0.000000


- Percent Identity ?? 
- Query Cover ??
- E value ??

add pictures
Homo sapiens = Human
Pan troglodytes = Chimpanzee
Gorilla gorilla gorilla = Western lowland gorilla
Pongo pygmaeus = Bornean orangutan

Interpretation :
- The APOE protein sequence was analyzed using the NCBI BLASTp tool against the non‑redundant protein database (nr).
- The results showed high sequence similarity among vertebrate species.
- The highest similarity was observed with primates such as chimpanzee and gorilla, indicating strong evolutionary conservation of the APOE protein.

- The results show a very high degree of conservation among primates, with over 97% sequence identity.
- This suggests that the APOE protein has remained largely unchanged throughout primate evolution.

### Graphic Summary

<img src="Images/graphic_summary.png" width="700">

### Distance Tree of Results


<img src="Images/Images/blast tree.png" width="700">

### Alignment